### PR TITLE
Event API

### PR DESCRIPTION
**Pull request by alexhsamuel**
Event API is ready, as described in notes/api.md.  This should be enough to get started on the front end.

No tests, as you can see.  Maybe one of us will motivate to learn how to unit-test Flask APIs at some point.  But I've done some ad-hoc testing.

This PR is incremental on #12.  Up to you if you prefer to review PRs one by one, or all changes together here.

### CODE REVIEW

File: backend/locator/api.py [hunk 1]
```diff
@@ -1,12 +1,109 @@
...
 
+class APIError(Exception):
+
+    def __init__(self, message, status_code=400):
+        self.message = message
+        self.status_code = status_code
+
+
+
+def jsonify(jso, status_code=200):
```
Comment (Kelvinm) [inline backend/locator/api.py]: this is fine, but Flask provides a Response class & object that allow you to do just this (Response(resp, status_code)) in place of jsonify

File: backend/locator/api.py [hunk 2]
```diff
@@ -30,54 +116,110 @@ def parse_date(string):
...
+
 @API.route("/events", methods=["GET"])
 def get_events():
-    return flask.jsonify({
-        "event_id": None,
-    })
+    cfg = flask.current_app.config
+    
+    # Start with all undeleted events.
+    query = SESSION.query(Event).filter_by(deleted=False)
```
Comment (bereli) [inline backend/locator/api.py]: may be by default we should filter on start_date > today

Comment (alexhsamuel) [reply backend/locator/api.py]: That's sort of a UI concern, isn't it?
Also, just being pedantic, "today" is time zone dependent so not the same for all clients.

**Comment by alexhsamuel** [comment]
@Kelvinm I don't see your comment...?

File: backend/locator/api.py (hunk 2 above)
Comment (bereli) [reply backend/locator/api.py]: that might be too much data to return by default, but that's out of scope of the current project.

**Comment by alexhsamuel** [comment]
Oh I see it now; GitHub marked it as outdated.

**Comment by alexhsamuel** [comment]
Hmm `flask.Result({ ... }, status_code)` doesn't seem to format the payload to JSON.

**Comment by Kelvinm** [comment]
More something along these lines:

http://flask.pocoo.org/docs/1.0/api/#response-objects
(many examples) https://www.programcreek.com/python/example/51515/flask.Response

return Response(json.dumps(task.dictize()), 200, mimetype='application/json')

**Pull request merged**